src/App.tsx

```diff
@@ -15,35 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetically = 'alphabetically',
+  byLength = 'byLength',
+  default = '',
+}
+
+type SortObj = {
+  sortField: SortType;
+  isReversed?: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed = false }: SortObj,
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.alphabetically:
+        return good1.localeCompare(good2);
+
+      case SortType.byLength:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibledGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortField(SortType.default);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.alphabetically,
+          })}
+          onClick={() => setSortField(SortType.alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.byLength,
+          })}
+          onClick={() => setSortField(SortType.byLength)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
```

Comment:
To enhance user experience, consider marking the 'Reverse' button when the list is reversed. This could be through a change in style or an icon, in accordance with the [UX] checklist.